src/mailbox-state.c

```diff
@@ -162,7 +162,12 @@ message_metadata_set_flags(struct imap_client *client, const struct imap_arg *ar
 		if (*atom == '\\') {
 			/* system flag */
 			flag = mail_flag_parse(atom + 1);
-			if (flag != 0)
+			if (client->imap4rev2_enabled &&
+			    (flag & MAIL_RECENT) != 0) {
+				imap_client_input_error(client,
+					"Invalid system flag: %s", atom);
```

Comment:
Since this is the same error, a bit easier would be to just set flag=0 and let the existing error handling log it